File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int mx = 0;
        int mn = prices[0];
        for(int i=1; i<n; i++){
            int c = prices[i]-mn;
            mx = max(c, mx);
            mn=min(mn, prices[i]);
        }
        return mx;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mini = prices[0];
        int maxi = 0;
        for(int i = 0; i<prices.length; i++){
            int diff = prices[i]-mini;
            maxi = Math.max(maxi, diff);
            mini = Math.min(mini, prices[i]);
        }
        return maxi;
    }
}